bigint.go

```diff
@@ -9,10 +9,14 @@ import (
 
 // ToBigInt converts a protobuf BigInt message to a *big.Int.
 func ToBigInt(b *protov1.BigInt) (*big.Int, error) {
+	if b == nil || len(b.GetValue()) == 0 {
+		return nil, errors.New("input cannot be nil or empty")
+	}
+
 	result := new(big.Int)
 	result = result.SetBytes(b.GetValue())
 
-	if len(result.Bits()) == 0 {
+	if result.Sign() == 0 {
```

Comment:
This does not handle negative numbers like previously. 

```suggestion
        if result.Sign() != 1 {
```

Reply:
ohh nice catch thanks @adelowo , even `!=1` also won't this condition only allows positive numbers and incorrectly handles any non-positive values, including zero, as errors.

Reply:
`if result.Sign() <= 0` right?